emptied repo for code review

lindsim opened the pull request.

lindsim committed:
emptied repo

joeyyang commented on docs/directions.js:
```diff
@@ -1,27 +0,0 @@
-    var convert = ["blank", "one", "two", "three", "four", "five", "six"];
-
-  $(document).ready(function () {
-    $("#directText").hide();
```
create a variable called `var $directText = $('#directText');` then refer to the var rather than calling the selector every time, same for every other selector you use — `$templates`, `$choices`, etc.

joeyyang replied:
`$directText.hide()`

joeyyang commented on docs/manipulation.js:
```diff
@@ -1,94 +0,0 @@
-  var previousShape = null;
-
-
-  $(".shape").mouseover(function(event) {
```
`var $shape` here — the `$` in the variable is a convention used to signify a jQuery element

joeyyang commented on docs/manipulation.js:
```diff
@@ -1,94 +0,0 @@
-      
-       switch(e.which) {
-          //space bar  
-          case 32:
```
In general, make this code read more like English. Do this by assigning variables to the cases rather than explaining which case is which through comments, and create submethods to say what each case does. Example:

```
var spaceBar = 32;
...
var keyPressed = e.which;
case spacebar: 
  changeColor(currentShape);
...

var changeColor = function (currentShape) {
  currentShape.removeClass(colorClasses).addClass(colorArray[Math.floor(Math.random()*11)]);
}
``` 
etc.

lindsim replied:
I renamed some divs and variables to make them more consistent too.

joeyyang commented on docs/manipulation.js:
```diff
@@ -1,94 +0,0 @@
-          
-          //left arrow
-          case 37:    
-            if (currentIndex > 0){
```
```
case leftArrow:
  rotateLeft(currentShape);
```

joeyyang commented on docs/directions.js:
```diff
@@ -1,27 +0,0 @@
-$(function () {
-
-    var currentTemplate = $("#blank");
-    var currentChoice = $("#0");
```
`$currentChoice`

joeyyang commented on docs/directions.js:
```diff
@@ -1,27 +0,0 @@
-$(function () {
-
```
```
this.registerSelectors();

this.registerSelectors = function() {
  this.$currentTemplate = $('#blank');
  this.$currentChoice = $('#0');
  ...
};

lindsim replied:
why is this superior to the way it is now?  feel free to tell me in person if it is a lengthly explanation.

joeyyang replied:
My opinion is that you want someone who is at the top of the file who is reading it for the first time to understand within a couple seconds what is happening in the file. Here, the first thing the file does is to register the jQuery selectors. I would make the top of the file read:

```
this.registerSelectors();
this.doStep2();
this.doStep3();
...
```

joeyyang commented on docs/index.html:
```diff
@@ -1,64 +0,0 @@
-    </section>
-    <main id="matchSpace">
-        <div id="directText">
-        <p>The object of the puzzle is to fit all of the shape pieces
```
- Make directions visual
- start with an outline (triangle) to strongly suggest dragging the pieces into the outline
- get rid of directions menu — replace with prompt like "fit the shapes into the outline!"

joeyyang commented on tangrams/sass/main.scss:
```diff
@@ -1,137 +0,0 @@
-  color: $black;
-}
-
-
```
get rid of blanklines

lindsim committed:
changed directions style per joey's suggestion

lindsim committed:
Merge branch 'master' into refactor

lindsim committed:
fix problem with showing and hiding outline at start

lindsim committed:
fix problem with showing and hiding outline at start

lindsim committed:
fix problem with showing and hiding outline at start

lindsim committed:
modified jquery to work with load-specific choices

lindsim committed:
Merge branch 'refactor' into code_review

Merged for code review with Joey.

lindsim committed:
Fixed spacing issue

lindsim committed:
Added semicolons. 

Note to self: Look up semicolon use with JavaScript and jQuery. Do semicolons matter more in strict mode?
